### PR TITLE
feat: API integration for login and registration page in admin app

### PR DESCRIPTION
**Pull request by sidarth-23**
## Summary
- Add form submission handling for the registration form to send a POST request to the REGISTER_URL with user data, set cookies on success, and handle errors from the server response
- Add form submission handling for the login form to send a POST request to the LOGIN_URL with user data, set cookies on success
- fix: update BASE_URL to default to `http://127.0.0.1:8000/api/` if not provided. The env is not reading properly. Hence the hard coded value for the time being for dev setup
- feat: add types for registration success and error responses, login success and error responses
- fix: layout shift in reset password page

### Checklist
- [x] Tested (Must)
- [ ] Test Case added
- [x] Build Successful (Must)
- [x] Sufficient Code comments added (Must)
- [ ] Attached Screenshots / Videos 
- [x] All Relevant Documents added

## Depends on
#77

### CODE REVIEW

**Review by yogai53 — DISMISSED** [review]

File: apps/admin/app/(auth)/login/_components/form.tsx [hunk 1]
```diff
@@ -24,8 +33,24 @@ export function LoginForm() {
...
+      .post<TLoginSuccessResponse>(LOGIN_URL, data)
+      .then((response) => {
+        document.cookie = `Authorization=${response.data.token.access}`;
+        document.cookie = `Refresh=${response.data.token.refresh}`;
+        router.push("/dashboard");
+      })
+      .catch((error: AxiosError<TLoginErrorResponse>) => {
+        console.log("error", error);
+        if (error.response) {
+          console.log("error.response", error.response);
```
Comment (yaju09) [inline apps/admin/app/(auth)/login/_components/form.tsx]: is the console required?

File: apps/admin/app/(auth)/login/_components/form.tsx [hunk 2]
```diff
@@ -24,8 +33,24 @@ export function LoginForm() {
...
+  const submitFunction = (data: TFormSchema) => {
+    axios
+      .post<TLoginSuccessResponse>(LOGIN_URL, data)
+      .then((response) => {
+        document.cookie = `Authorization=${response.data.token.access}`;
+        document.cookie = `Refresh=${response.data.token.refresh}`;
+        router.push("/dashboard");
+      })
+      .catch((error: AxiosError<TLoginErrorResponse>) => {
+        console.log("error", error);
```
Comment (yaju09) [inline apps/admin/app/(auth)/login/_components/form.tsx]: is the console required?

File: apps/admin/app/(auth)/register/_components/form.tsx [hunk 3]
```diff
@@ -21,8 +28,44 @@ export function RegisterationForm() {
...
+    axios
+      .post<TRegistrationSuccessResponse>(REGISTER_URL, data)
+      .then((response) => {
+        document.cookie = `Authorization=${response.data.token.access}`;
+        document.cookie = `Refresh=${response.data.token.refresh}`;
+        router.push("/dashboard");
+      })
+      .catch((error: AxiosError<TRegistrationErrorResponse>) => {
+        if (error.response) {
+          error.response.data.email &&
```
Comment (yaju09) [inline apps/admin/app/(auth)/register/_components/form.tsx]:
```suggestion
         const error =  error.response.data
         error.email && form.setError("email", {
              type: "server",
              message: error.email[0],
            });
         ```
         is it possible to reduce the size of this function

**Review by yaju09 — CHANGES_REQUESTED** [review]

File: apps/admin/app/(auth)/register/_components/form.tsx (hunk 3 above)
Comment (sidarth-23) [reply apps/admin/app/(auth)/register/_components/form.tsx]: For the time being until we come up with a consistent naming scheme for frontend and backend, it is not possible to reduce the function size

**Review by yaju09 — APPROVED** [review]

**Review by elizabeththomas92 — APPROVED** [review]

**Pull request merged**